packages/block-editor/src/hooks/block-renaming.js

```diff
@@ -1,78 +0,0 @@
-/**
- * WordPress dependencies
- */
-import { addFilter } from '@wordpress/hooks';
-import { hasBlockSupport } from '@wordpress/blocks';
-import { __ } from '@wordpress/i18n';
-import { TextControl } from '@wordpress/components';
-
-/**
- * Internal dependencies
- */
-import { InspectorControls } from '../components';
-
-/**
- * Filters registered block settings, adding an `__experimentalLabel` callback if one does not already exist.
- *
- * @param {Object} settings Original block settings.
- *
- * @return {Object} Filtered block settings.
- */
-export function addLabelCallback( settings ) {
-	// If blocks provide their own label callback, do not override it.
-	if ( settings.__experimentalLabel ) {
-		return settings;
-	}
-
-	const supportsBlockNaming = hasBlockSupport(
-		settings,
-		'renaming',
-		true // default value
-	);
-
-	// Check whether block metadata is supported before using it.
-	if ( supportsBlockNaming ) {
-		settings.__experimentalLabel = ( attributes, { context } ) => {
-			const { metadata } = attributes;
-
-			// In the list view, use the block's name attribute as the label.
-			if ( context === 'list-view' && metadata?.name ) {
-				return metadata.name;
-			}
-		};
-	}
-
-	return settings;
-}
```

Comment:
I believe the bug @jasmussen spotted with the group block is because this code needs to remain.

I've made a separate PR now that removes just the control, so I think we should be able to consider it a completely separate task:
[Remove block renaming control from advanced inspector controls group](https://github.com/WordPress/gutenberg/pull/60453#top)